examples/ampi/cuda/simple/Makefile

```diff
@@ -15,10 +18,16 @@ CXX=$(AMPICXX)
 LD=$(AMPICXX)
 CXXFLAGS=$(DEFS)
 
-simple: simple-cuda.o
+simple: simple-cuda.o simple.o
+	$(AMPICXX) -o $@ simple.o simple-cuda.o $(LD_LIBS)
 
 simple-cuda.o: simple-cuda.cu
-	$(NVCC) $(DEFS) -std=c++11 -I$(CHARMDIR)/include -c simple-cuda.cu
+	$(NVCC) $(NVCC_INC) $(DEFS) -std=c++11 -I$(CHARMDIR)/include -c simple-cuda.cu
+
+simple.o: simple.C
+		$(AMPICXX) -c $<
```

Comment:
The `$(AMPICXX)` line here without `$(CHARMC_INC)` can be removed as redundant.

Reply:
> Does this also fix the segfault described in #2074 or is that pending a separate PR?

Shixing built this on his laptop, and when we went to run it he got an abort during startup from CUDA saying that his device driver was too old and was incompatible with the CUDA runtime library. So we need to test it somewhere else

Reply:
I managed to get it running on my laptop and tried to run the test in the Makefile. It failed the last test.
```
../../../../bin/testrun  +p2 ./simple +vp 16 
Charmrun> scalable start enabled. 
ssh: connect to host localhost port 22: Connection refused
Charmrun> Error 255 returned from remote shell (localhost:0)
Charmrun> Reconnection attempt 1 of 3
ssh: connect to host localhost port 22: Connection refused
Charmrun> Error 255 returned from remote shell (localhost:0)
Charmrun> Reconnection attempt 2 of 3
ssh: connect to host localhost port 22: Connection refused
Charmrun> Error 255 returned from remote shell (localhost:0)
Charmrun> Reconnection attempt 3 of 3
ssh: connect to host localhost port 22: Connection refused
Charmrun> Error 255 returned from remote shell (localhost:0)
Charmrun> Too many reconnection attempts; bailing out
Makefile:33: recipe for target 'test' failed
make: *** [test] Error 1
'''

Reply:
That's not the test itself failing, that's the test failing to run because it could not recursively and non-interactively ssh into the your machine. You'll need to set up a public/private key pair that can be used to ssh into localhost without needing a password. Alternatively, run `make test TESTOPTS="++local"` to skip the need for ssh altogether.

Reply:
I managed to get the program running on my laptop and it passed the first three tests in the makefile './simple +vp 1''./simple +vp 2''./simple +vp 16' and also passed 'charmrun ./simple +p4 ++local +vp16'.

Reply:
Yes I was running the program locally without external nodes. 
Here's the output of the some tests:
```
charmrun ./simple +p4 ++local +vp16
Charmrun> scalable start enabled. 
Charmrun> started all node programs in 0.026 seconds.
Charm++> Running in non-SMP mode: 4 processes (PEs)
Converse/Charm++ Commit ID: v6.9.0-374-g670ae1bf7
Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
Charm++> scheduler running in netpoll mode.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (1 sockets x 4 cores x 2 PUs = 8-way SMP)
Charm++> cpu topology info is gathered in 0.029 seconds.
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
[Partition 0][Node 0] End of program
```
```
./simple +vp4
Charm++: standalone mode (not using charmrun)
Charm++> Running in non-SMP mode: 1 processes (PEs)
Converse/Charm++ Commit ID: v6.9.0-374-g670ae1bf7
Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
Charm++> scheduler running in netpoll mode.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (1 sockets x 4 cores x 2 PUs = 8-way SMP)
Charm++> cpu topology info is gathered in 0.000 seconds.
Calling kernel...
Calling kernel...
Calling kernel...
Calling kernel...
[Partition 0][Node 0] End of program
```

Reply:
The requested changes are resolved.